lib/src/rules/avoid_returning_null_for_future.dart

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2018, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:analyzer/dart/element/type.dart';
+import 'package:linter/src/analyzer.dart';
+
+const _desc = r'Avoid returning null for Future.';
+
+const _details = r'''
+
+**AVOID** returning null for Future.
+
+Future being used to represent a potential value, or error, that will be
+available at some time in the future it's almost always wrong to return null for
```

Comment:
This description is a little hard to follow...  Maybe just simplify it to:

> It is almost always wrong to return null for a `Future`. Most of the time the developer simply forgot to put an `async` keyword on the function.

Reply:
Done